### DISCUSSION TITLE
v2.1

### DISCUSSION THREAD

**Discussion by eduardofilo** [eduardofilo/RG350_adam_image#276] [Announcements]
## Installation instructions

* [In english](https://github.com/eduardofilo/RG350_adam_image/wiki/En:-2.-Install-&-Upgrade#fresh-installation): It is mandatory to follow the instructions or the console will not boot up.
* [En español](https://github.com/eduardofilo/RG350_adam_image/wiki/Es:-2.-Instalaci%C3%B3n-y-Actualizaci%C3%B3n#instalaci%C3%B3n-desde-cero): Es obligatorio seguir las instrucciones o la consola no arrancará.

## Upgrading instructions

If you upgrade from version 2.0 to 2.1, the following items will be preserved as they are kept on the external card since 2.0:

* Savestates from standalone emulators and RetroArch.
* ROM preferences from SimpleMenu.
* Favorites from SimpleMenu.

If you upgrade from a 1.x branch version to 2.1, follow the upgrade instructions in the [2.0 release](https://github.com/eduardofilo/RG350_adam_image/releases/tag/v2.0).

## Change log

* ODbeta rolled back to 2022-02-13 (5.15 version).
* RetroArch updated to 2022-07-14 (1.10.3 version).

<hr /><em>This discussion was created from the release <a href='https://github.com/eduardofilo/RG350_adam_image/releases/tag/v2.1'>v2.1</a>.</em>

**Comment by nlboom** [comment]
@eduardofilo Hi. Why you rolled back to old ODbeta?

**Reply by eduardofilo** [reply]
Because I have received reports of various problems in v2.0 that have made me think that it is better to use a system that has been published longer and has been tested more by users.
These are probably things that can be explained, but I prefer to use the precautionary principle. Anyway, you can update ODbeta on your own, and if you do it and find that it doesn't give problems or even improves some things, I'd appreciate it if you'd let us know.

**Comment by OptiZonion** [comment]
I'm sorry if this sounds like a silly question but how does one actually upgrade from 2.0 to 2.1 ? The "Upgrading Instructions" section doesn't really list the procedure. Where should the image be copied and how to trigger the update ? Is it a card flash again ?

Thank you.

**Reply by eduardofilo** [reply]
It is commented on in this section, but perhaps I was not very explicit. The update is as usual by flashing the new version on the internal card. The difference with previous versions is that before you had to backup the game savestates and now you don't, as they are kept on the external card. I will try to rephrase that explanation to make it clearer.

https://github.com/eduardofilo/RG350_adam_image/wiki/En:-2.-Install-&-Upgrade#upgrade-from-v20

**Reply by kaossino** [reply]
Hi, I use v2.0 and I changed some Retroarch and Sound mixer settings, when I update to 2.1 will I lose these differences?

**Comment by RCCRF** [comment]
CAN YOU ACTUALLY TURN OFF THE DEVICE NOW? Since god forbid you try to on the old versions. JUST would reset the os, the sleep function never worked either. SO when I'd be done playing, I'd literally have to let the battery run down and flip over the device since god forbid you could turn off the display even.

**Reply by eduardofilo** [reply]
I'm not sure I understand the comment. I've never had a problem shutting down the console in this or previous versions. I don't think anyone else has because I'm sure I would have had numerous notifications about it. The sleep function also works. Sounds like it's a problem with your particular unit. There are some marginal machines that produce all sorts of weird behaviour.

**Reply by RCCRF** [reply]
ON THE 350M when I try to power down the device from RETROARCH, it will never power down, it will restart via the POWER button after pressing it for what feels like forever. It will never shut down. Issue also plagues me in some of the stand alone emulators.

**Reply by eduardofilo** [reply]
Please describe, or better make a video, how you try to shut down the console from RetroArch. By holding down the Power key? By showing the RetroArch shutdown menu which is hidden by default? Other methods?

**Reply by chrisstonehill** [reply]
Did this get resolved? I had the never ending reboot issue, seemingly after I tried using the hdmi out. Eduardo you had a video showing a way to resolve this endless reboot cycle...but I forget the details...and maybe I actually resolved my problem from someone's comment before I saw the video...involving loooooong press of the reset button, maybe...like a hard reset or something. I have't tried hdmi out since then.

**Comment by MAPesh** [comment]
Hey there, what's the difference between the two dump files? Sorry for silly question, just getting into this.

**Reply by eduardofilo** [reply]
Sorry for the late reply. That clarification was indeed missing. I thought it was detailed in the wiki but it was not. I have already added it to this section:
https://github.com/eduardofilo/RG350_adam_image/wiki/En:-2.-Install-&-Upgrade#fresh-installation

**Comment by ogramilik** [comment]
i can't install the latest version of adam on my rg350p please help me :/

**Reply by eduardofilo** [reply]
Sorry for the late reply. Check if in your case, as is happening to many RG350P owners, you have a high resolution panel. In that case select the RG350M model in the kernel selection script as commented here:
https://github.com/eduardofilo/RG350_adam_image/wiki/En:-6.-Problem-solving#console-does-not-boot-after-flashing

**Reply by ogramilik** [reply]
mine happens another problem, there is a message on the screen of "Generating ASA key..." and it does not come out of it :/
![IMG_20221215_112545](https://user-images.githubusercontent.com/108706386/207885950-23f46154-c48a-4618-9007-2910096dd9b0.jpg)

**Reply by eduardofilo** [reply]
I would need more information to get to the ultimate problem, but it has happened one time in the past on my machine and I seem to remember it was related to problems with the state of the file system on one of the two cards, I can't quite remember. Try checking the internal one on a Linux machine and the external one on a Linux or Windows machine. Or reinstall everything, preferably on a different card, especially the internal one.

**Reply by ogramilik** [reply]
I will do that, thanks in advance for your attention ☺️

**Reply by ogramilik** [reply]
Take advantage of this and ask me a question, today the new 32gb netac micro sd I bought arrived and now I'm going to test it again as soon as I install the system again if the laptop will exit the game and return to the game selection screen, this happened, I was told that the card I was using was of poor quality, now I have 2 netacs of 32gb both bought in the official netac store on Aliexpress, will the problem stop?

**Reply by ogramilik** [reply]
And about arcade games, I can never put it on, do you know the correct procedure?  I have to download the game and in which of the folders?  because it has arcade, mame, among others and it is the first time that I will put arcade games on it.  and have to download 1 or more bios?  and why in which specific folder?

**Reply by eduardofilo** [reply]
Try this guide. If there is something you don't understand, this will help me to improve it.
https://github.com/eduardofilo/RG350_adam_image/wiki/En:-3.-Content-installation

**Reply by JazzJohannes** [reply]
@eduardofilo this guide helped me too, it was simply using the select kernel switch for 350m instead of 350p, since they ran out of the low-res displays in the actual hardware, therefore 350m was working amazingly.

**Reply by ogramilik** [reply]
Hello again, sorry to bother you again but I'm just going to report a problem that happened to me, I hope it's a very isolated case and not because of some system bug, I was playing my rg350p and normally when the battery is at 15% o laptop shows the red light then I put it to recharge immediately but after using the first time on the adam 2.1 it simply did not notify the low battery at 15% and the light remained green and out of nowhere it turned off and as soon as I recharged and turned it on again the games were gone , I went to open the card via pc and I had an abcd.iso file inside and after formatting what was before a 32gb card now only shows 2gb, I tried everything formatting using my high quality ugreen 3.0 adapter both on a smartphone and on a computer and none of them solved it, could it be that adam himself, when disconnecting, permanently damaged my card?  I'm pretty upset :(

**Reply by eduardofilo** [reply]
It sounds like the card has definitely broken. I don't think it was due to the abrupt disconnection when the battery died, but anything can happen. As for the battery indicator, it is not infallible, as it is a completely analogue system that varies from machine to machine. That's why I wrote [this topic](https://github.com/eduardofilo/RG350_adam_image/wiki/En:-7.-FAQ#q11-why-does-not-show-battery-levels-correctly-the-battery-indicator).
Just don't rush the battery so much when you see that your unit behaves like this. Frankly, I don't think there is anything wrong with the system (which, as I always say, I don't develop). These are simply not carrier grade machines with simple electronics and software (precisely because they have a simple processor).

**Comment by hyvr** [comment]
Hello, everything is working with this version for me on the 280V, except for the NES/Famicom which shows the "opendgx" loading screen and then crashes back to game selection. I have flashed this on two different 280V devices to try, and also 2.0 version, with the same results. Is it my roms or something else I can fix?

**Reply by BuscandoMemo** [reply]
I have the same issue. I was assuming it was user error since I got my device yesterday until I saw you were commenting with the same issue. I have tried one rom from an alternate source so far with no luck. And I have tried with the files uncompressed.

**Reply by eduardofilo** [reply]
Hi, I can assure you that this system has always worked. Just in case I have tested on a clean install of Adam and all the NES games I have tried work. I have tried both zip compressed games and nes format directly. And with both cores available to emulate that machine. Maybe you have compressed ROMs with a "codec" that is not able to decompress the RetroArch engine. Try decompressing to `nes` format. Try also with the alternative core. By default FCEUmm is used, but you also have QuickNES.

**Reply by BuscandoMemo** [reply]
Hello, thank you for your reply. In my case it was my specific ROM files. Tried some different ROMs from different sources and it is working as designed now. Thanks!

**Comment by VampireODBC** [comment]
Hi @eduardofilo, I'm new here. I've just ordered a new RG280V from a shop and they already installed Adam Image. If I want to update it to 2.1 then all what I must do is just simply flashing the 2.1 image to my device. Am I correct here?

**Reply by eduardofilo** [reply]
Yes, follow this guide:
https://github.com/eduardofilo/RG350_adam_image/wiki/En:-2.-Install-&-Upgrade#fresh-installation

By the way, where did you buy the unit with Adam preinstalled if I can ask?

**Reply by VampireODBC** [reply]
Wow thanks for the fast reply. 
I'm living in Vietnam. In my country there's a console shop name Topo Shop. Their technicians preloaded the Adam Image before selling the RG280V to customers. They also do creating ROM collections and localization. 
I'm not sure which version installed on my device (which is on delivery). I'll check it out and upgrade my device following your guide whenever I received the package. And, thanks for the awesome work!

**Reply by VampireODBC** [reply]
Hi, it's me again. I got the device, the default hotkeys worked perfectly (L3 + R3 to open menu, Select + R1 to quicksave,...), but unfortunately I messed up with my mistake. The config was reset and now I can't set up to get those hotkeys work again. I can't find any solution, can you send me the cfg file of RetroArch :(? How dumb I am to mess it up

**Reply by eduardofilo** [reply]
Here it is: https://raw.githubusercontent.com/eduardofilo/RG350_adam_image/master/data/local/home/.retroarch/retroarch.cfg

**Reply by VampireODBC** [reply]
Thanks, I tried copying the file to its folder but the buttons stop working :(

**Reply by VampireODBC** [reply]
Mine is RG280V, and only the Select + Start works 😅

**Reply by eduardofilo** [reply]
Check if there is any file inside the `autoconfig` directory (I speak from memory, the name might change slightly) inside the RetroArch home and if so delete it.

**Reply by VampireODBC** [reply]
Thanks man, I figured out the problem now. Now the hotkeys works lke a charm. Enjoying my retro NES games 😁

**Comment by JazzJohannes** [comment]
Commodore 64 and AMIGA do not work in this release!

**Reply by chrisstonehill** [reply]
Did they work on older releases? I can't get Amiga to work on 1.3.1. Planning to try the latest version in the next few weeks and was hoping that would result in some Amiga success. I have picked a more focussed bunch of roms, from a few thousand.

**Comment by Rabobert** [comment]
i updated from 2.0 but unfortunatly, i have an issue now with brightness. It boots up well but half way through the boot the brightness turns down extremely. Any fix for this? Thanks!
